### DISCUSSION TITLE
Sentinel Policies support?

### DISCUSSION THREAD

**Discussion by suprjinx** [openbao/openbao#783] [General]
Hi! I'm wondering if there's any roadmap plan to implement "Sentinel Policies" from Vault Enterprise?

**Comment by cipherboy** [comment]
\o hey @suprjinx -- we had some discussion on #514 about alternative auth methods. Sentinel is entirely a proprietary feature of Vault Enterprise, so we're unable to support it in a drop-in compatible way. Additionally it is very much a HashiCorp-specific solution. I'd like to extend the ACL policy (as you can see from #514 but also #769 and similar coming RFC around paginated list limitations) as much as possible, but I think ultimately we'll need to plugin-ize the auth method to get things like Cedar, Cels, OPA, or other custom solutions. 

If you're interested in working on something in that regard, would be happy to chat and collaborate!

**Reply by suprjinx** [reply]
I'd vote for embracing OPA/Rego as a Sentinel replacement -- it seems conceptually to address the same domain, and Terraform Enterprise offers it as an alternative to Sentinel.

**Reply by cipherboy** [reply]
The biggest item here that's unclear in my mind is how OPA would integrate into the auth/identity model. If you can put together a proposal for that, it'd be great to read :-)

**Reply by achetronic** [reply]
A first approach to have something to help making policies more versatile [here](https://github.com/openbao/openbao/pull/1260) 🙂

**Comment by alexandrust88** [comment]
any new updates here in regards if was set on any roadmap  release for it  ?

**Reply by cipherboy** [reply]
I'm happy to draft an RFC for CEL so we have one to work off of, theres been some very recent work on general plugins in #3547 that's also relevant.

**Comment by klaus-sap** [comment]
> ultimately we'll need to plugin-ize the auth method to get things like Cedar, Cels, OPA, or other custom solutions.

Proposal: RFC on External Authorization https://github.com/openbao/openbao/pull/4003
